superset/commands/database/csv_import.py

```diff
@@ -0,0 +1,164 @@
+    header_row: int
+    index_column: str
+    null_values: list[str]
+    overwrite_duplicates: bool
+    rows_to_read: int
+    skip_blank_lines: bool
+    skip_initial_space: bool
+    skip_rows: bool
+
+
+class CSVImportCommand(BaseCommand):
+    def __init__(
+        self,
+        model_id: int,
+        table_name: str,
+        file: Any,
+        options: CSVImportOptions,
+    ) -> None:
+        self._model_id = model_id
+        self._model: Optional[Database] = None
+        self._table_name = table_name
+        self._schema = options.get("schema")
+        self._file = file
+        self._options = options
+
+    def run(self) -> None:
+        self.validate()
+        if not self._model:
+            return
+        try:
+            df = pd.concat(
+                pd.read_csv(
+                    chunksize=READ_CSV_CHUNK_SIZE,
+                    encoding="utf-8",
+                    filepath_or_buffer=self._file,
+                    header=self._options.get("header_row", 0),
+                    index_col=self._options.get("index_column"),
+                    dayfirst=self._options.get("day_first", False),
+                    iterator=True,
+                    keep_default_na=not self._options.get("null_values"),
+                    usecols=self._options.get("columns_read")
+                    if self._options.get("columns_read")
+                    else None,
+                    na_values=self._options.get("null_values")
+                    if self._options.get("null_values")
+                    else None,
+                    nrows=self._options.get("rows_to_read"),
+                    parse_dates=self._options.get("column_dates"),
+                    sep=self._options.get("delimiter", ","),
+                    skip_blank_lines=self._options.get("skip_blank_lines", False),
+                    skipinitialspace=self._options.get("skip_initial_space", False),
+                    skiprows=self._options.get("skip_rows", 0),
+                    dtype=self._options.get("column_data_types")
+                    if self._options.get("column_data_types")
+                    else {},
+                )
+            )
+            csv_table = Table(table=self._table_name, schema=self._schema)
+            self._model.db_engine_spec.df_to_sql(
+                self._model,
+                csv_table,
+                df,
+                to_sql_kwargs={
+                    "chunksize": READ_CSV_CHUNK_SIZE,
+                    "if_exists": self._options.get("already_exists", "fail"),
+                    "index": self._options.get("index_column"),
+                    "index_label": self._options.get("column_labels"),
+                },
+            )
+        except ValueError as ex:
+            raise DatabaseUploadFailed(
+                message=_(
+                    "Table already exists. You can change your"
+                    " 'if table already exists' strategy to append or replace."
```

Comment:
Is another option here for them to rename the table. Should we mention that @yousoph?

Reply:
Table already exists. You can change your 'if table already exists' strategy to append or replace or provide a different Table Name to use.